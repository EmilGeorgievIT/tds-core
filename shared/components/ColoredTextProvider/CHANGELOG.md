# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="1.0.0"></a>
# 1.0.0 (2018-12-06)


### Features

* **shared:** add tag property to ColoredTextProvider ([e77b659](https://github.com/telus/tds-core/commit/e77b659))
* **shared-colored-text-provider:** create new private package ([813b6fa](https://github.com/telus/tds-core/commit/813b6fa))
