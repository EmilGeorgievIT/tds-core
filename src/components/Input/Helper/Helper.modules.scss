@import '../../../scss/settings/colours';
@import '../../../scss/settings/spacing';

$border-radius: 4px;

.base {
  padding: $spacing-base;
  margin-bottom: $spacing-tight; // TODO: This should be moved up to be consistent with form field spacing

  border-radius: $border-radius;

  // transition: background-color .1s linear; TODO: Why?
}

.default {
  composes: base;

  background: $color-athens-grey;
}

.success {
  composes: base;

  background: $color-panache;
}

.error {
  composes: base;

  background-color: $color-lavender-blush;
}

// TODO: This is duplicated in Notification...
.errorText {
  color: $color-cardinal;
}
