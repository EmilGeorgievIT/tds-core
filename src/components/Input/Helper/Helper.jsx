import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'
import ColoredTextProvider from '../../Typography/ColoredTextProvider/ColoredTextProvider'

import messagingStyles from '../../Messaging.modules.scss'
import styles from './Helper.modules.scss'

const getClassName = feedback => (feedback ? styles[feedback] : styles.default)

const renderContent = (feedback, children) => {
  if (feedback === 'error') {
    return (
      <ColoredTextProvider colorClassName={messagingStyles.errorText}>
        {children}
      </ColoredTextProvider>
    )
  }

  return children
}

const Helper = ({ feedback, children, ...rest }) => (
  <div {...safeRest(rest)} className={getClassName(feedback)}>
    {renderContent(feedback, children)}
  </div>
)
Helper.propTypes = {
  feedback: PropTypes.oneOf(['success', 'error']),
  children: PropTypes.node.isRequired
}
Helper.defaultProps = {
  feedback: undefined
}
Helper.displayName = 'Input.Helper'

export default Helper
