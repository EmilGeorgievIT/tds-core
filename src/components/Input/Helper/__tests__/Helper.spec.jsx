import React from 'react'
import { shallow, render } from 'enzyme'
import toJson from 'enzyme-to-json'

import ColoredTextProvider from '../../../Typography/ColoredTextProvider/ColoredTextProvider'

import Helper from '../Helper'

describe('Helper', () => {
  const defaultChildren = 'Some helper text.'
  const doShallow = (props = {}, children = defaultChildren) => (
    shallow(<Helper {...props}>{children}</Helper>)
  )
  const doRender = (props = {}, children = defaultChildren) => (
    render(<Helper {...props}>{children}</Helper>)
  )

  it('renders', () => {
    const helper = doRender()

    expect(toJson(helper)).toMatchSnapshot()
  })

  it('can have a feedback state', () => {
    let helper = doShallow()
    expect(helper).toHaveClassName('default')

    helper = doShallow({ feedback: 'success' })
    expect(helper).toHaveClassName('success')
  })

  it('does not color the success content', () => {
    const helper = doShallow({ feedback: 'success' }, 'A success message')

    expect(helper.find(ColoredTextProvider)).not.toBePresent()
  })

  it('colors the error content', () => {
    const helper = doShallow({ feedback: 'error' }, 'An error message')

    expect(helper).toContainReact(
      <ColoredTextProvider colorClassName="errorText">
        An error message
      </ColoredTextProvider>
    )
  })

  it('passes additional attributes to the element', () => {
    const helper = doShallow({ id: 'the-helper', role: 'alert' })

    expect(helper).toHaveProp('id', 'the-helper')
    expect(helper).toHaveProp('role', 'alert')
  })

  it('does not allow custom CSS', () => {
    const helper = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(helper).not.toHaveProp('className', 'my-custom-class')
    expect(helper).not.toHaveProp('style')
  })
})
