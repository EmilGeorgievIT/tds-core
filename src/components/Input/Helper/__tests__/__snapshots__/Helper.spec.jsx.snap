// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper renders 1`] = `
<div
  class="default"
>
  Some helper text.
</div>
`;
